packages/runtime-common/realm.ts

```diff
@@ -883,10 +778,7 @@ export class Realm {
     try {
       // local requests are allowed to query the realm as the index is being built up
       if (!isLocal) {
-        // allow any WIP index requests to query the index while it's building up
-        if (!request.headers.get('X-Boxel-Use-WIP-Index')) {
-          await this.ready;
-        }
```

Comment:
turns out I was using the loader wrong. the realm's loader has the internal fetch URL handler. once I started passing a clone of that to the worker then this became unnecessary